llvm/lib/Transforms/Yk/BasicBlockTracer.cpp

```diff
@@ -51,12 +55,31 @@ struct YkBasicBlockTracer : public ModulePass {
         continue;
       }
       uint32_t BlockIndex = 0;
-      // FIXME: Once control point transition is implemented,
-      //        only add tracing calls to unopt version.
       for (auto &BB : F) {
         builder.SetInsertPoint(&*BB.getFirstInsertionPt());
-        builder.CreateCall(TraceFunc, {builder.getInt32(FunctionIndex),
-                                       builder.getInt32(BlockIndex)});
+
+        if (ModuleClone) {
+          bool isUnopt = F.getName().startswith(YK_SWT_UNOPT_PREFIX);
+          bool isAddrTaken =
+              F.getMetadata(YK_SWT_MODCLONE_FUNC_ADDR_TAKEN) != nullptr;
+
+          if (isAddrTaken || isUnopt) {
+            // Address-taken and unoptimised functions are not cloned during
```

Comment:
I don't understand this comment. Can we try rewording this? e.g why are unopt functions not cloned? What _are_ unopt functions? Why does not cloning them mean they need to be executed during tracing and what are real trace calls (as opposed to unreal trace calls)?

Reply:
Updated :point_right: 6a785b14318b8829d5cf2a79b55128d19dbb47c3